### PR TITLE
Add unit tests with dot and dot-dot patterns. Improve doublestar.FilepathGlob() logic

### PR DESCRIPTION
**Pull request by sebastien-rosset**
Fix issue #68 

I've noticed `doublestar.FilepathGlob` has the following behavior:

1. Both `filepath.Glob(".")` and `doublestar.FilepathGlob("..")` return `[]string{"."}`. This is what I expected.
1. `filepath.Glob("..")` returns `[]string{".."}` but `doublestar.FilepathGlob("..")` returns `nil`. I was expecting  `doublestar.FilepathGlob("..")` to also return `[]string{".."}`. Do you agree?

This relates to the question you had in my previous PR: https://github.com/bmatcuk/doublestar/pull/61/files#r917288214

I've added unit tests to show the potential problem. Also, here is a short program showing the behavior with `filepath.Glob()`: https://go.dev/play/p/MhdYounCCMP

I fixed the problem by:
1. Improving the comparison between `doublestar.FilepathGlob()` and `filepath.Glob()`.
2. Improve the `doublestar.FilepathGlob()` logic when a pattern includes `.`, `..` or `/`.

### CODE REVIEW

File: utils.go [hunk 1]
```diff
@@ -88,6 +88,11 @@ func FilepathGlob(pattern string) (matches []string, err error) {
 		// of forward slashes, no matter what the system uses
 		matches[i] = filepath.FromSlash(path.Join(base, matches[i]))
 	}
+	if len(matches) == 0 {
```
Comment (sebastien-rosset) [inline utils.go:91]: Without this code, the new unit tests would fail.

**Comment by sebastien-rosset** [comment]
@bmatcuk would you be able to review?

**Comment by sebastien-rosset** [comment]
Gently trying to get this pull request reviewed. Thank you.

**Comment by bmatcuk** [comment]
@sebastien-rosset thanks for the PR! And sorry it's taken so long to get back to it! Where does the time go?

I've come up with an [alternative implementation](https://github.com/bmatcuk/doublestar/pull/76). I realized all of your test cases can just short-circuit; no need to actually try to glob anything. I also put these special-case tests into their own file so as to not complicate the main tests.

Let me know what you think!

**Comment by sebastien-rosset** [comment]
Thanks!! The alternative looks good to me.

**Pull request closed without merging**